server/schedule/placement/fit.go

```diff
@@ -186,7 +187,15 @@ func (w *fitWorker) run() {
 // Index specifies the position of the rule.
 // returns true if it replaces `bestFit` with a better alternative.
 func (w *fitWorker) fitRule(index int) bool {
+	if w.exit {
+		return false
+	}
 	if index >= len(w.rules) {
+		// If there is no isolation level and we already find one solution, we can early exit searching instead of
+		// searching the whole cases.
+		if w.bestFit.IsSatisfied() && !needIsolation(w.rules) {
```

Comment:
I'd like to propose to calculate `needIsolation` when creating the worker to save CPU when `needIsolation` is always true.

Reply:
good catch, updated.